src/base-commerce-command.js

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2021 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const { Command } = require('@oclif/command')
+const { cli } = require('cli-ux')
+const { initSdk } = require('./cloudmanager-helpers')
+class BaseCommerceCommand extends Command {
+  async runSync (programId, environmentId, body, pollingInterval, imsContextName = null) {
+    const sdk = await initSdk(imsContextName)
+    const postResponse = await sdk.postCLICommand(programId, environmentId, body)
+
+    const { id: commandId } = await postResponse
+
+    let result = await this.callGet(sdk, programId, environmentId, commandId)
+
+    while (result.status === 'RUNNING' || result.status === 'CREATING') {
+      await cli.wait(pollingInterval)
+      result = await this.callGet(sdk, programId, environmentId, commandId, true)
+    }
+    cli.action.stop(`\nMESSAGE: ${result.message}`)
+    return result
+  }
+
+  async callGet (sdk, programId, environmentId, commandId) {
+    const getResponse = await sdk.getCLICommand(programId, environmentId, commandId)
+
+    const result = await getResponse
+    cli.action.start(`STATUS: ${result.status}`)
```

Comment:
I generally get the impression that `cli.action.start()` and `cli.action.stop()` are intended to be called in pairs. IIUC the way this is structured, there could be N number of `start()` calls. It does though seem to work. But to be clear, the result of this is effectively _replacing_ the output from any prior `cli.action.start` calls. Is that actually what is desired?

What I would suggest instead is to call `cli.action.start` with ONE argument *before* the POST request. And then make subsequent calls in this method with the same first argument and the status as the second argument. Because it is temporal, hard to show in a github comment, but the effect would be:

Time 0 (prior to POST): Running <command>... <spinner>
Time 1 (after the first GET): Running <command>... <spinner> <status>
Time 2 (after the first GET): Running <command>... <spinner> <status>
Time N (once the command has completed): Running command... <message>

Reply:
Understand the desire to put the status at the beginning of the output. Do think it is weird that this isn't an option in cli-ux, but definitely not something we should be trying to subclass or address here.